stdpopsim/slim_engine.py

```diff
@@ -948,17 +948,58 @@ def fix_time(event):
     condition_on_allele_frequency = []
     op_id = stdpopsim.ext.ConditionOnAlleleFrequency.op_id
     slim_mutation_ids = []
-    for x in _dfe_to_mtypes(contig).values():
+    slim_mutation_dfe = []
+    for i, x in _dfe_to_mtypes(contig).items():
         slim_mutation_ids.extend([u[0] for u in x])
+        slim_mutation_dfe.extend([i for u in x])
     for ee in extended_events:
-        if hasattr(ee, "mutation_type_id"):
-            mt_id = getattr(ee, "mutation_type_id")
+        if hasattr(ee, "single_site_id"):
+            mut_id = getattr(ee, "single_site_id")
             cls_name = ee.__class__.__name__
-            if mt_id not in slim_mutation_ids:
+            mtype_idx = [
+                j
+                for j, i in enumerate(slim_mutation_dfe)
+                if contig.dfe_list[i].id == mut_id
+            ]
+            n_mtypes = len(mtype_idx)
+            if n_mtypes != 1:
                 raise ValueError(
-                    f"Invalid {cls_name} event with mutation type id "
-                    f"{mt_id} on {contig} with valid IDs {slim_mutation_ids}."
+                    f"The single site mutation type with id '{mut_id}' "
+                    f"referenced by {cls_name} must exist and be uniquely "
+                    f"labelled, but there are {n_mtypes} other mutation types "
+                    f"with this id on {contig}. "
                 )
+            mtype_idx = mtype_idx[0]
+            setattr(ee, "mutation_type_id", slim_mutation_ids[mtype_idx])
+            dfe_idx = slim_mutation_dfe[mtype_idx]
+            dfe_intervals = contig.interval_list[dfe_idx]
+            if dfe_intervals.shape[0] == 0:
+                raise ValueError(
+                    f"The single site id '{mut_id}' referenced by {cls_name} "
+                    f"has no coordinate: it may have been removed by the addition "
+                    f"of an overlapping DFE or a single site with an "
+                    f"identical coordinate."
```

Comment:
```suggestion
                    f"has no coordinate: it may have been removed by the addition "
                    f"of an overlapping DFE or a single site with an "
                    f"identical coordinate after instatiation of '{cls_name}'."
```

Reply:
Hm, not sure I understand-- instantiation of `DrawMutation` won't check that the single site exists/has a coordinate -- this is done in `slim_makescript`. So one could add a single site, add a DFE overlapping the single site, then setup the extended events; and the simulation would still raise an error.

Reply:
IIUC, Andy is just trying to make the error message more clear. Currently, a user might think from the error message that first adding the overlapping DFE and then adding the single site would produce this error, so he's clarifying that it's only a problem if the overlapping DFE is added to the contig *after* the single site.

Reply:
yes indeed @petrelharp -- thanks for clarifying my cryptic stuff